backend/auth/role_map_repository.go

```diff
@@ -15,7 +16,8 @@ type RoleMapRepository struct {
 	// Subrole map is optional, if not provided, subroles will be ignored. Roles received in token are checked only with RoleMap
 	RoleMap map[string]*models.Role
 	SubroleMap map[string]*models.Role
-	flattenedMap map[string]map[string]map[string][]models.OperationType
+	flattenedMap map[string]map[string]map[string]map[models.OperationType]struct{}
```

Comment:
This is a nightmare fuel.

Reply:
you can think of it as a map of role -> permission matrix

Reply:
I had no nightmares about that while coding so can't confirm this is a valid comment

Reply:
It looks like a nightmare, but under the hood it's brilliant. I really admire you for finding this optimization